### PR TITLE
Convert 15 packages from embedded go gets to go/bump.

### PR DESCRIPTION
**Pull request by dlorenc**

### CODE REVIEW

File: nfs-subdir-external-provisioner.yaml [hunk 1]
```diff
@@ -21,19 +21,14 @@ pipeline:
       tag: nfs-subdir-external-provisioner-${{package.version}}
       expected-commit: c2a2d5d544781e3d3e4d7a7e2d21a8e64cf6d9d1
 
-  # I added `go mod edit -go=1.17` because when I do `go get golang.org/x/net@v0.7.0,`
-  # to fix a High CVE it gave me an error unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod.
+  - uses: go/bump
+    with:
+      deps: github.com/prometheus/client_golang@v1.11.1 golang.org/x/text@v0.9.0 golang.org/x/net@v0.17.0 gopkg.in/yaml.v3@v3.0.0-20220521103104-8f96da9f5d5e
```
Comment (imjasonh) [inline nfs-subdir-external-provisioner.yaml:26]: Minor UX feedback, I think we should standardize on this for better readability/diffs

```
deps: >-
  github.com/prometheus/client_golang@v1.11.1
  golang.org/x/text@v0.9.0 golang.org/x/net@v0.17.0
  gopkg.in/yaml.v3@v3.0.0-20220521103104-8f96da9f5d5e
```

Comment (joshrwolf) [inline nfs-subdir-external-provisioner.yaml:26]:
```suggestion
      deps: >-
        github.com/prometheus/client_golang@v1.11.1
        golang.org/x/text@v0.9.0
        golang.org/x/net@v0.17.0
        gopkg.in/yaml.v3@v3.0.0-20220521103104-8f96da9f5d5e
```

File: prometheus-adapter.yaml [hunk 2]
```diff
@@ -20,17 +20,11 @@ pipeline:
...
-
-      # Mitigate CVE-2023-47108
-      go get go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.46.0
-      go get go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.46.0
-      go get go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc@v1.21.0
-      go get go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp@v1.21.0
-      go mod tidy
+  - uses: go/bump
+    with:
+      deps: golang.org/x/net@v0.17.0 go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.46.0 go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.46.0 go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc@v1.21.0 go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp@v1.21.0
```
Comment (joshrwolf) [inline prometheus-adapter.yaml:25]:
```suggestion
      deps: >-
        golang.org/x/net@v0.17.0
        go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.46.0
        go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.46.0
        go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc@v1.21.0
        go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp@v1.21.0
```

File: nfs-subdir-external-provisioner.yaml (hunk 1 above)
Comment (joshrwolf) [reply nfs-subdir-external-provisioner.yaml:26]: talking to jason, lets ignore this and fix any styling in follow ups

**Review by joshrwolf — DISMISSED** [review]

**Review by ajayk — APPROVED** [review]

**Pull request merged**